Microsoft.DotNet.Interactive/Extensions/KernelExtensions.cs

```diff
@@ -23,38 +37,39 @@ public static class KernelExtensions
                 throw new ArgumentNullException(nameof(onVisit));
             }
 
-
-            if (kernel is ICompositeKernel compositeKernel)
+            foreach (var subKernel in KernelHierarchy.GetChildren(kernel))
             {
-                foreach (var subkernel in compositeKernel.ChildKernels)
-                {
-                    onVisit(subkernel);
+                onVisit(subKernel);
 
-                    if (recursive &&
-                        subkernel is ICompositeKernel childCompositeKernel)
-                    {
-                        childCompositeKernel.VisitSubkernels(onVisit, true);
-                    }
+                if (recursive)
+                {
+                    subKernel.VisitSubKernels(onVisit, recursive: true);
                 }
             }
         }
 
-        public static async Task VisitSubkernelsAsync(
+        public static async Task VisitSubKernelsAsync(
```

Comment:
The casing change is incorrect. The `K` should be lower-case.

Reply:
fixing it